keydata_test.go

```diff
@@ -76,7 +76,29 @@ func (h *mockPlatformKeyDataHandler) checkState() error {
 	}
 }
 
+func (h *mockPlatformKeyDataHandler) validateFields(data *PlatformKeyData) error {
+	if data.Version < 0 || data.Version == 1 || data.Version > 2 {
+		return xerrors.Errorf("Invalid KeyData version: %d", data.Version)
+	}
+
+	if data.Version == 2 {
+		if data.KDFAlg < crypto.MD4 || data.KDFAlg > crypto.BLAKE2b_512 {
+			return xerrors.Errorf("Unexpected KDFAlg: %d", data.KDFAlg)
+		}
+	}
+
+	if data.AuthMode < AuthModeNone || data.AuthMode > AuthModePassphrase {
+		return xerrors.Errorf("Invalid AuthMode: %d", data.AuthMode)
+	}
+
+	return nil
```

Comment:
The validation should check that the supplied fields are the expected values rather than bounds checking them. I'm not sure of the best way to do this though - maybe adding the expected values as fields to `mockPlatformKeyDataHandle`?

Reply:
Reverted the last attempt and pushed some changes to the logic if you could take a look. I added `Expected*` fields to the `mockPlatformKeyDataHandle` but tbh I find it a bit confusing to parse.

Reply:
I think this is ok. One other way to implement this would be to make `mockKeyDataHandler` use GCM and have the extra properties as part of the AAD instead, but maybe we can change that in a follow-up.